drivers/espi/host_subs_npcx.c

```diff
@@ -474,77 +487,85 @@ static void host_pmch_ibf_isr(const void *arg)
 
 /* Host port80 sub-device local functions */
 #if defined(CONFIG_ESPI_PERIPHERAL_DEBUG_PORT_80)
-static void host_port80_isr(const void *arg)
+#if defined(CONFIG_ESPI_NPCX_PERIPHERAL_DEBUG_PORT_80_MULTI_BYTE)
+static void host_port80_work_handler(struct k_work *item)
 {
-	ARG_UNUSED(arg);
-	struct shm_reg *const inst_shm = host_sub_cfg.inst_shm;
-	struct espi_event evt = { ESPI_BUS_PERIPHERAL_NOTIFICATION,
-		(ESPI_PERIPHERAL_INDEX_0 << 16) | ESPI_PERIPHERAL_DEBUG_PORT80,
-		ESPI_PERIPHERAL_NODATA
-	};
-	uint8_t status = inst_shm->DP80STS;
-
-	if (!IS_ENABLED(CONFIG_ESPI_NPCX_PERIPHERAL_DEBUG_PORT_80_MULTI_BYTE)) {
-		LOG_DBG("%s: p80 status 0x%02X", __func__, status);
-
-		/* Read out port80 data continuously if FIFO is not empty */
-		while (IS_BIT_SET(inst_shm->DP80STS, NPCX_DP80STS_FNE)) {
-			LOG_DBG("p80: %04x", inst_shm->DP80BUF);
-			evt.evt_data = inst_shm->DP80BUF;
+	uint32_t code = 0;
+	struct host_sub_npcx_data *data = CONTAINER_OF(item, struct host_sub_npcx_data, work);
+	struct ring_buf *rbuf = &data->port80_ring_buf;
+	struct espi_event evt = {ESPI_BUS_PERIPHERAL_NOTIFICATION,
+				 (ESPI_PERIPHERAL_INDEX_0 << 16) | ESPI_PERIPHERAL_DEBUG_PORT80,
+				 ESPI_PERIPHERAL_NODATA};
+
+	while (!ring_buf_is_empty(rbuf)) {
+		struct npcx_dp80_buf dp80_buf;
+		uint8_t offset;
+
+		ring_buf_get(rbuf, &dp80_buf.offset_code[0], sizeof(dp80_buf.offset_code_16));
```

Comment:
Can you use `&dp80_buf.offset_code_16` as the *data argument? That way you have the same field as dereference and sizeof, makes it more obvious that it's the correct size you are taking.

Reply:
The `ring_buf_get()` function expects the 2nd argument to be 'uint8_t *. Passing `&dp80_buf.offset_code_16` to it (without casting)  causes the compiler to complains of it.
So I changed that to ` ring_buf_peek(rbuf, &dp80_buf.offset_code[0], ARRAY_SIZE(dp80_buf.offset_code));` in the updated patch.
Is that change ok with you?

Reply:
Got it, no `ARRAY_SIZE` is number of elements so it should definitely not be there, use `sizeof(dp80_buf.offset_code)` instead, that returns the size of the whole array.

Reply:
Done.